src/components/ErrorPages/SessionExpired.tsx

```diff
@@ -12,27 +110,128 @@ export default function SessionExpired() {
     toast.dismiss();
   }, []);
 
+  const { seconds, breathState, timeStr, scaleFactor, shouldShowTime } =
+    useSessionTimer();
+
+  const renderSegmentedDigit = (digit: string) => {
+    const map = digitMaps[digit] || digitMaps["0"];
+
+    return (
+      <div className="inline-block mx-0.5">
+        {map.map((row, rowIndex) => (
+          <div key={rowIndex} className="flex">
+            {row.map((cell, cellIndex) => (
+              <div
+                key={cellIndex}
+                className={`w-1.5 h-1.5 m-px ${
+                  cell ? "bg-gray-400" : "bg-transparent"
+                }`}
+              ></div>
+            ))}
+          </div>
+        ))}
+      </div>
+    );
+  };
+
+  const renderSegmentedTime = (timeStr: string) => (
+    <div
+      className={cn("flex items-center justify-center transform origin-center")}
+      style={{ transform: `scale(${scaleFactor})` }}
+    >
+      {timeStr.split("").map((char, index) => (
+        <div key={index}>{renderSegmentedDigit(char)}</div>
+      ))}
+    </div>
+  );
```

Comment:
Do not inline component definitions inside another component! React will not be able to memoize this properly as on every re-render, the reference to the variable changes!